components/forms/Form/Form.tsx

```diff
@@ -151,70 +151,90 @@ const InnerForm: React.FC<InnerFormProps> = (props) => {
-                <p className="gc-description">{t("spam-error.success-prompt")}</p>
+          <form
+            id="form"
+            data-testid="form"
+            method="POST"
+            onSubmit={(e) => {
+              e.preventDefault();
+              if (timerActive) {
+                if (!formTimerState.canSubmit) {
+                  window.dataLayer = window.dataLayer || [];
+                  window.dataLayer.push({
+                    event: "form_submission_spam_trigger",
+                    formID: formID,
+                    formTitle: form.titleEn,
+                    submitTime: formTimerState.remainingTime,
+                  });
+                  setSubmitTooEarly(true);
+                  // In case the useEffect timer failed check again
+                  //formTimerDispatch({ type: "check" });
+                  return;
+                }
+                // Only change state if submitTooEarly is already set to true
+                submitTooEarly && setSubmitTooEarly(false);
+              }
+
+              if (isReCaptchaEnableOnSite) {
+                handleSubmitReCaptcha(e);
+              } else {
+                handleSubmit(e);
+              }
+            }}
+            noValidate
+          >
+            {children}
+            <div
+              className={classNames({
+                "border-l-2": submitTooEarly,
+                "border-red-default": submitTooEarly,
+                "border-green-default": formTimerState.remainingTime === 0 && submitTooEarly,
+                "pl-3": submitTooEarly,
+              })}
+            >
+              {submitTooEarly &&
+                (formTimerState.remainingTime > 0 ? (
+                  <div role="alert">
+                    <p className="gc-label text-red-default">
+                      {t("spam-error.error-part-1")} {formTimerState.timerDelay}{" "}
+                      {t("spam-error.error-part-2")}
+                    </p>
+                    <p className="gc-description">
+                      {t("spam-error.prompt-part-1")} {formTimerState.remainingTime}{" "}
+                      {t("spam-error.prompt-part-2")}
+                    </p>
+                  </div>
+                ) : (
+                  <div role="alert">
+                    <p className="gc-label text-green-default">{t("spam-error.success-message")}</p>
+                    <p className="gc-description">{t("spam-error.success-prompt")}</p>
+                  </div>
+                ))}
+
+              <RichText>
+                {form.privacyPolicy &&
+                  form.privacyPolicy[props.language == "en" ? "DescrptionEn" : "DescriptionFr"]}
+              </RichText>
+
+              <div className="buttons">
+                <Button type="submit" disabled={props.isPreview}>
+                  {t("submitButton")}
+                </Button>
+                {props.submitAlert && (
```

Comment:
I know this is already in but wondering if the naming here could be better  for `submitAlert `